pennylane_lightning/src/util/Util.hpp

```diff
@@ -181,6 +250,59 @@ inline auto log2(size_t value) -> size_t {
     return static_cast<size_t>(std::log2(value));
 }
 
+/**
+ * @brief Define popcount for multiple compilers as well as different types.
+ *
+ * TODO: change to std::popcount in C++20
+ */
+///@{
+#if defined(_MSC_VER)
+constexpr auto popcount(uint64_t val) -> int { return __popcnt(val); }
+#elif defined(__GNUC__) || defined(__clang__)
+constexpr auto popcount(unsigned long val) -> int {
```

Comment:
Can we have `int`, `uint64_t` and `unsigned long` replaced with `std::size_t`?

Reply:
I also would like to, but as builtin functions `__popcnt` (MSVC), `__builtin_popcount` (GCC or Clang) returns `int`, I think it's safer to keep it int.

Reply:
However, I also found that the recent version of MSVC raises integer conversion errors as its builtin function returns the same type as the input. Thus I anyway need to cast, so changed all return types as `size_t`.